test/diagnostics/import.jl

```diff
@@ -0,0 +1,279 @@
+    l = isnothing(m) ? l : "ecg $(m[1])"
+
+    m = match(r"[\s\[,]*e[ck]g([123])[\]\s,]*"i, l)
+    l = isnothing(m) ? l : "ecg $(m[1])"
+
+    # [LR] ECG
+    m = match(r"^[\[\s,]*(?<lr>[lr])\s*E[ck]g"i, l)
+    l = isnothing(m) ? l : "ecg av$(m[:lr])"
+
+    # [LR]-Leg[12]?
+    m = match(r"^[\[\s,]*(?<lr>[lr])\s*\-\s*leg[1-2][\s\],]?"i, l)
+    l = isnothing(m) ? l : "ecg av$(m[:lr])"
+
+    # EMG Leg
+
+    # {left,right,l,r}[\s-_]*leg
+    m = match(r"^(\s*emg\s*)?(left|l)[_\-\s]*leg\s*(?<rest>.*)\s*$"i, l)
+    l = isnothing(m) ? l : "emg left_anterior_tibialis$(m[:rest])"
+    m = match(r"^(\s*emg\s*)?(right|r)[_\-\s]*leg\s*(?<rest>.*)\s*$"i, l)
+    l = isnothing(m) ? l : "emg right_anterior_tibialis$(m[:rest])"
+
+    # leg[\s-_]*{left,right,l,r}
+    m = match(r"^(\s*emg\s*)?(?<lr>left|l)[_\-\s]*leg\s*(?<rest>.*)\s*$"i, l)
+    l = isnothing(m) ? l : "emg left_anterior_tibialis$(m[:rest])"
+    m = match(r"^(\s*emg\s*)?(?<lr>right|r)[_\-\s]*leg\s*(?<rest>.*)\s*$"i, l)
+    l = isnothing(m) ? l : "emg right_anterior_tibialis$(m[:rest])"
+
+    # (Tib|Leg)(-|/)[LR]
+    m = match(r"^\s*(leg|tib)[/\-](?<lr>l|left|right|r)\s*$"i, l)
+    l = isnothing(m) ? l : "$(startswith(m[:lr], "l") ? "left" : "right")_anterior_tibialis"
+
+    # [LR]Leg1 - [LR]Leg2
+    m = match(r"[\s\[,]*(?<lr>[lr])leg([12]?)[\s\-lr]*leg([12]?)[\]\s,]*"i, l)
+    l = isnothing(m) ? l : "emg $(m[:lr])leg"
+
+    # [LR] LEG[12]?  # with a minus `-`, this would be interpreted as ECG
+    m = match(r"^[\[\s,]*(?<lr>[lr])\s*leg[1234]?"i, l)
+    l = isnothing(m) ? l : "emg leg$(m[:lr])"
+
+    return l
+end
+
+custom_labels = deepcopy(OndaEDF.STANDARD_LABELS)
+# add alternative pap signal name
+custom_labels[["positive_airway_pressure", "xpap"]] = ["ipap", "epap", "cpap"]
+#custom_labels[["emg"]] = [p for p in custom_labels[["emg"]] if startswith(first(p), "chin")]
+
+custom_extractors = [edf -> extract_channels_by_label(edf, signal_names, channel_names; preprocess_labels=test_preprocessor)
+                     for (signal_names, channel_names) in custom_labels]
+
+function has_leg(h)
+    h.kind != "emg" && return false
+    return any(c -> startswith(c, "left_anterior_tibialis") || startswith(c, "right_anterior_tibialis"), h.channels)
+end
+
+@testset "Import EDF" begin
+
+    @testset "edf_to_samples_info" begin
+        results = map(test_edf_to_samples_info) do r
+            edf = mock_edf(r)
+            return last(OndaEDF.edf_header_to_onda_samples_info(edf; custom_extractors=custom_extractors))
+        end
+        # print result of mapping `edf_to_onda_samples` over `test_edf_to_samples_info.in`
+        # this makes it easy to see effects of any changes made to OndaEDF by looking at
+        # `diff test_edf_to_samples_info.{in,out}`
+        #
+        # in practice this doesn't work well because the ordering of dicts is not defined
+        # so the header map entries print out in different orders.
```

Comment:
so this is just testing that the code actually runs without error and not content?

Re ordering of dicts: you could define the ordering behavior to be based on `sort(keys(dict))` for consistency

Reply:
none of the code in this directory is actually run anymore, just preserved for posterity

Reply:
(and to  make it a bit easier if we want to restore the tests later)

Reply:
I had completely lost the thread by this point. :smile:

Reply:
also (now that I'm thining about it again) because I wrote a bit of glue code to convert the old diagnostics into a table and vice-versa that might be useful.

Reply:
If these tests aren't being run at all then I would delete them. They're available in the git history if we ever want to restore them but in the meantime it's confusing to have a bunch of bitrotten tests lying around unused.